### PR TITLE
Some minor improvements in table

### PR DESCRIPTION
**Pull request by raphaelsc**

### CODE REVIEW

File: replica/database.hh [hunk 1]
```diff
@@ -426,10 +426,6 @@ private:
     // Ensures that concurrent updates to sstable set will work correctly
     seastar::named_semaphore _sstable_set_mutation_sem = {1, named_semaphore_exception_factory{"sstable set mutation"}};
     mutable row_cache _cache; // Cache covers only sstables.
-    // FIXME: until we use uuid for sstables generation (#10459)
```
Comment (tchaikov) [inline replica/database.hh:429]: i think the "FIXME" still holds unless #10459 is fixed.

File: replica/database.hh [hunk 2]
```diff
@@ -426,10 +426,6 @@ private:
...
-    // _sstable_generation keeps track of the highest
-    // generation number in the table so that newly created sstables
-    // will use numeric generation numbers larger than this one.
```
Comment (tchaikov) [inline replica/database.hh:432]: i think, this comment might be still valid as long as we s/_sstable_generation/_sstable_generation_generator/ .

Comment (raphaelsc) [reply replica/database.hh:432]: the comment is actually no longer valid. all the logic was pushed to the generator, and there's nothing left to be done here in replica::table.

**Comment by raphaelsc** [comment]
@scylladb/scylla-maint ping for merge.

**Pull request closed without merging**